analyzers/gradle/gradle.go

```diff
@@ -179,20 +181,26 @@ func (a *Analyzer) Analyze() (graph.Deps, error) {
 	} else {
 		if a.Options.Project != "" {
 			project = a.Options.Project
-		} else {
+		} else if config.Version() < 2 {
```

Comment:
I think I'd be more comfortable seeing this factored out into a function. You'll repeat more code this way, but it'll be easier to reason about since version checks won't be peppered throughout the code.

For example, I'd prefer:

```
func parseModuleV1() {
	// logic here
}

func parseModuleV2() {
	// logic here, a lot of this will look very similar
}

func (a *Analyzer) Analyze() {
	// ...
	switch config.Version() {
		case 2: parseModuleV2()
		case 1: parseModuleV1()
	}
	//...
}
```

Reply:
I agree with this approach. While I think this will add a significant amount of lines it will reduce the amount of time to understand whats happening.